### PR TITLE
Application Insights components & webtests API schema updates

### PR DESCRIPTION
**Pull request by d3r3kk**
## Description
Update Application Insights webtest and components API to version 2015-05-01

## Requirements:
- [x] PR should pass the validateJSON.js
- [x] PR should pass runSchemaValidation.js
- [x] PR should pass runSchemaTests.js
- [x] The PR should include at least one test for each resource type added in this CR.
    - Added two tests each, no tests existed before
- [x] Respect coding style of 2 space indentation for JSON files.
- [x] Update schemas\2014-04-01-preview\deploymentTemplate.json
- [x] Update schemas\2015-01-01\deploymentTemplate.json

### CODE REVIEW

File: schemas/2015-05-01/microsoft.insights.json [hunk 1]
```diff
@@ -0,0 +1,657 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2015-05-01/microsoft.insights.json#",
```
Comment (TianoMS) [inline schemas/2015-05-01/microsoft.insights.json]: >microsoft.insights [](start = 64, length = 18)

please be consistent in names. First letter upper case, so: "Microsoft.Insights".

File: schemas/2015-05-01/microsoft.insights.json [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the Application Insights component resource."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "microsoft.insights/components"
```
Comment (TianoMS) [inline schemas/2015-05-01/microsoft.insights.json]: >microsoft.insights [](start = 13, length = 18)

These one as well: first letter in upper case. Please fix all of them.

**Comment by d3r3kk** [comment]
Updated naming and improved the schema overall by removing some extra, unneeded definitions.
- [Improved schema generation from swagger source, PR here](https://github.com/Azure/azure-rest-api-specs/pull/2632)

**Comment by d3r3kk** [comment]
@TianoMS, any further suggestions?

**Comment by Tiano2017** [comment]
@d3r3kk can you fix the file name as well? it's "microsoft.insights.json" now.

**Comment by d3r3kk** [comment]
@TianoMS you bet. All done.

**Pull request merged**